### ISSUE TITLE
Add tests for Boilerplate generated code

### ISSUE THREAD

**Issue by InTheNow** [non/cats#383]
Code generated from `Boilerplate.scala` should be tested by generated tests.

**Comment by BennyHill** [comment]
Just a heads up that a new version of scoverage is likely to pick up the generated code and, unless these classes are removed from coverage, you are likely to see _BIG_ drop in code coverage.

As to why you should test generated code, its more that you want test the code that _it_ calls, in the correct context.

**Comment by non** [comment]
Agreed -- this will definitely become a pressing issue. I think the suggestion to generate corresponding tests makes sense.
